src/Components/map.jsx

```diff
@@ -1,50 +1,384 @@
-import React, { useEffect, useState } from 'react';
-import { MapContainer, TileLayer, Polyline } from 'react-leaflet';
-import 'leaflet/dist/leaflet.css';
-import axios from 'axios';
-import { polyline } from 'leaflet';
+import React, { useEffect, useState } from "react";
+import {
+  MapContainer,
+  TileLayer,
+  Polyline,
+  Marker,
+  useMap,
+  useMapEvents,
+  Popup,
+} from "react-leaflet";
+import L from "leaflet";
+import "leaflet/dist/leaflet.css";
+import axios from "axios";
+import "../index.css";
 
-export default function App (){
```

Comment:
You might want to rename this function because there is a standard App function that is used in React projects and this should not be used for anything outside of the standard use case.